### PR TITLE
Add CO2STORE + GASWAT case

### PR DESCRIPTION
**Pull request by goncalvesmachadoc**

### CODE REVIEW

**Comment by totto82** [comment]
Thanks. I tried to stripped down version(replaced compositional input with GAS/WATER/DISGASW) and removed some output.  The case run after implementing support for WSF/GSF relperms. But the pressure is increasing significantly. Does it make sense to add boundary conditions to the test case? Either an aquifer or just a large porevolume multiplier on the edge.  

image


image

**Comment by negar-geo** [comment]
you can add pore volume multiplier to the left and right boundary, then the pressure doesnt increase fast, so I applied this 
EDIT
----------------------------------------------------------------------------

MULTIPLY 
PORV 10000000 1 1 1 50 1 10/ 
PORV 10000000 50 50 1 50 1 10/ 
/
 and the pressure change in reservoir is as shown below 

pressure

**Comment by bska** [comment]
@goncalvesmachadoc: Do you expect to make further changes to this test case?  If not, we should probably try to get the model into the main repository.

File: co2store/CO2STORE_GASWAT.DATA [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+RPTSOL
+ PRESSURE SGAS SWAT DENG DENW VGAS VWAT XMF  YMF WMF /
+
+SUMMARY  ===========================================================
+
+ALL
+
+WXMF
+'*' 1 /
+'*' 2 /
```
[the remark this answers is not in the document]
Comment (bska) [reply co2store/CO2STORE_GASWAT.DATA:152]: Oops, sorry, no.  That doesn't work.  The component index (`1` and `2` in this example) will generate errors in the summary node creation.  You'd have to add some logic to the [`keywordW()`](https://github.com/OPM/opm-common/blob/03ab6356fb82e4244512f8bd94de92fc7c18f77e/src/opm/input/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp#L668) handler in `SummaryConfig.cpp` to intercept the mole fraction keywords (and do nothing).

Additionally, you'd have to create a new `_PROBE` keyword collection, maybe `WELL_COMPONENT_PROBE`, possibly started from a copy of [`WELL_COMPLETION_PROBE`]() that lists the `WXMF`, `WYMF`, and `WZMF` summary keywords.  I can assist you if you'd like.

[the remark this answers is not in the document]
Comment (goncalvesmachadoc) [reply co2store/CO2STORE_GASWAT.DATA:152]: Thank you for the detailed answer. Think will get the parsing of the summary keywords in a different PR than https://github.com/OPM/opm-common/pull/3442 then.

**Comment by goncalvesmachadoc** [comment]
image

**Comment by bska** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  I'll merge this into the master branch now.
